server/api/build.go

```diff
@@ -34,6 +38,53 @@ import (
 	"github.com/woodpecker-ci/woodpecker/server/store"
 )
 
+func CreateBuild(c *gin.Context) {
+	_store := store.FromContext(c)
+	repo := session.Repo(c)
+
+	var p woodpecker.BuildOptions
+
+	err := json.NewDecoder(c.Request.Body).Decode(&p)
+	if err != nil {
+		_ = c.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	user := session.User(c)
+
+	lastCommit, _ := server.Config.Services.Remote.BranchHead(c, user, repo, p.Branch)
+
+	tmpBuild := createTmpBuild(model.EventManual, lastCommit, repo, user, &p)
+
+	build, err := pipeline.Create(c, _store, repo, tmpBuild)
+	if err != nil {
+		handlePipelineErr(c, err)
+	} else {
+		c.JSON(http.StatusOK, build)
+	}
+}
+
+func createTmpBuild(event model.WebhookEvent, commitSHA string, repo *model.Repo, user *model.User, opts *woodpecker.BuildOptions) *model.Build {
+	return &model.Build{
+		Event:     model.EventManual,
+		Commit:    commitSHA,
+		Branch:    opts.Branch,
+		Timestamp: time.Now().UTC().Unix(),
+
+		Avatar:  user.Avatar,
+		Message: "MANUAL BUILD @ " + opts.Branch,
```

Comment:
Would still like to change this to the last commit message of the corresponding branch, but we could refactor this in separate PR together with the remote apis.